### PR TITLE
Enhance Managed Entity

### PR DESCRIPTION
**Pull request by pagbabian-splunk**
#### Related Issue: N/A
Managed Entity was too open ended, without enough detail on the changes to an arbitrary entity as a string value in `type`.  This PR adds specific object entities.

#### Description of changes:
Added `type_id` (along with existing `type` as sibling) with an initial set of types: Device, User, Group, Organization, Policy.
Added to the constraints as `at_least_one`

### CODE REVIEW

File: objects/managed_entity.json [hunk 1]
```diff
@@ -15,12 +15,65 @@
...
+      "requirement": "recommended"
+    },
+    "policy": {
+      "requirement": "recommended",
+      "description": "Describes details of a managed policy."
+    },
     "uid": {
       "description": "The identifier of the managed entity."
     },
+    "user": {
```
Comment (mikeradka) [inline objects/managed_entity.json:60]: What is the difference between the `user` and an entity that is of type `user`? Could we add clarification in the this description of `user`?

Comment (pagbabian-splunk) [reply objects/managed_entity.json:60]: When `type_id` = 3 (User) then the `user` attribute is populated.  They go together.

Comment (pagbabian-splunk) [reply objects/managed_entity.json:60]: We certainly can add better instructions (and ultimately, validators could check to match up which attribute is populated (just_one from a set, not the name/uid which would be a different set) for a particular type_id.

**Comment by mikeradka** [comment]
Looks pretty good to me. Once it is out of draft, I would be willing to re-review and approve.

**Review by mikeradka — APPROVED** [review]

File: objects/managed_entity.json [hunk 2]
```diff
@@ -15,12 +15,72 @@
       "description": "The managed entity type. For example: <code>policy</code>, <code>user</code>, <code>organizational unit</code>, <code>device</code>.",
       "requirement": "recommended"
     },
+    "type_id": {
+      "requirement": "recommended",
```
Comment (cmcginley-splunk) [inline objects/managed_entity.json:19]: Should `type_id` instead be required, with an additional enum for `Other`? I feel like that's the paradigm I've seen more frequently in other OCSF definitions.

Comment (pagbabian-splunk) [reply objects/managed_entity.json:19]: The issue with making it Required is that it would be a breaking change.  And yes, 0/99 will be included.

**Review by cmcginley-splunk — APPROVED** [review]

**Review by zschmerber — APPROVED** [review]

**Review by Aniak5 — APPROVED** [review]
LGTM!

**Pull request merged**